src/xmldiffng/main2.py

```diff
@@ -0,0 +1,262 @@
+        list_print(added[0], 'FILE:', 'added')
+        list_print(added[1], 'DIRECTORY:', 'added')
+        exit_status += sum((len(deleted[0]), len(deleted[1]),
+                            len(added[0]), len(added[1])))
+        for filename in common[0]:
+            print '-'*80
+            print 'FILE:', filename
+            diffs = process_files(os.path.join(fpath1, filename),
+                                  os.path.join(fpath2, filename),
+                                  norm_sp, xupd, ezs, verbose,
+                                  ext_ges, ext_pes, include_comment,
+                                  encoding, html)
+            if diffs:
+                exit_status += diffs
+    # if  args are files
+    elif os.path.isfile(fpath1) and os.path.isfile(fpath2):
+        if prof:
+            import profile, pstats, time
+            from maplookup import fmes_end,fmes_init, fmes_node_equal, has_couple, match_end, partner, lcs2
+            import maplookup
+            # replaces cfunction in maplookup by python wrappers
+            def fmes_end_w(*args):
+                return fmes_end(*args)
+            maplookup.fmes_end = fmes_end_w
+
+            def fmes_init_w(*args):
+                return fmes_init(*args)
+            maplookup.fmes_init = fmes_init_w
+
+            def fmes_node_equal_w(*args):
+                return fmes_node_equal(*args)
+            maplookup.fmes_node_equal = fmes_node_equal_w
+
+            def has_couple_w(*args):
+                return has_couple(*args)
+            maplookup.has_couple = has_couple_w
+
+            def match_end_w(*args):
+                return match_end(*args)
+            maplookup.match_end = match_end_w
+
+            def partner_w(*args):
+                return partner(*args)
+            maplookup.partner = partner_w
+
+            def lcs2_w(*args):
+                return lcs2(*args)
+            maplookup.lcs2 = lcs2_w
+
+            t = time.clock()
+            profiler = profile.Profile()
+            profiler.runctx('process_files(%r,%r,%r,%r,%r,%r,%r,%r,%r,%r,%r)'% (
+              fpath1, fpath2, norm_sp, xupd, ezs, verbose, ext_ges, ext_pes,
+              include_comment, encoding, html), globals(), locals() )
+            profiler.dump_stats( prof )
+            print 'Time:',`time.clock()-t`
+            p = pstats.Stats(prof)
+            p.sort_stats('time','calls').print_stats(.25)
+            p.sort_stats('cum','calls').print_stats(.25)
+
+        else:
+            exit_status = process_files(fpath1, fpath2,
+                                        norm_sp, xupd, ezs, verbose,
+                                        ext_ges, ext_pes, include_comment,
+                                        encoding, html)
+    else:
+        exit_status = -1
+        print fpath1, 'and', fpath2, \
+              'are not comparable, or not directory nor regular files'
+    sys.exit(exit_status)
+
+if __name__ == '__main__':
```

Comment:
We have that already in a way. ;-) I think you should merge it with `src/xmldiffng/cli.py`. There is a `main()` function already, so better to use this.